lib/saml_idp/request.rb

```diff
@@ -115,10 +115,11 @@ def valid_signature?
     end
 
     def service_provider?
-      service_provider.valid?
+      service_provider && service_provider.valid?
```

Comment:
Or implement it correctly if you prefer and then request doesn't need to change at all.